test/target_web/controllers/api/v1/session_controller_test.exs

```diff
@@ -1,46 +1,50 @@
 defmodule TargetWeb.API.V1.SessionControllerTest do
   use TargetWeb.ConnCase
 
-  alias Pow.Ecto.Schema.Password
-  alias Target.{Repo, Users.User}
   alias TargetWeb.APIAuthPlug
 
   @pow_config [otp_app: :target]
 
-  setup %{conn: conn} do
-    user =
-      Repo.insert!(%User{
-        email: "test@example.com",
-        password_hash: Password.pbkdf2_hash("secret1234")
-      })
-
-    {:ok, conn: conn, user: user}
-  end
-
   describe "create/2" do
-    @valid_params %{"user" => %{"email" => "test@example.com", "password" => "secret1234"}}
+    @valid_credentials %{"email" => "test@example.com", "password" => "secret1234"}
+    @valid_params %{"user" => @valid_credentials}
     @invalid_params %{"user" => %{"email" => "test@example.com", "password" => "invalid"}}
 
-    test "with valid params", %{conn: conn} do
+    test "with valid params and confirmed email", %{conn: conn} do
+      create_user(@valid_credentials, :confirmed)
       conn = post(conn, Routes.api_v1_session_path(conn, :create, @valid_params))
 
       assert json = json_response(conn, 200)
       assert json["data"]["token"]
       assert json["data"]["renew_token"]
     end
 
-    test "with invalid params", %{conn: conn} do
+    test "with valid params and unconfirmed email", %{conn: conn} do
```

Comment:
The same goes for these. part of doing tests is using them as documentation. Use cases should be clear when looking at them